pkg/controller/controller_test.go

```diff
@@ -2456,11 +2459,47 @@ func assertNumEvents(t *testing.T, strings []string, number int) {
 	}
 }
 
+// failfFunc is a type that defines the common signatures of T.Fatalf and
+// T.Errorf.
+type failfFunc func(t *testing.T, msg string, args ...interface{})
+
+func fatalf(t *testing.T, msg string, args ...interface{}) {
+	t.Log(string(debug.Stack()))
+	t.Fatalf(msg, args...)
```

Comment:
I appreciate the difference in style and readability. I think in this case, I want to hear what @arschles has to say. As I stated before, he's historically been opposed to this pattern because it makes the source of test failures more difficult to identify. I can see both sides of it.